swap/src/protocol/bob/event_loop.rs

```diff
@@ -228,22 +208,22 @@ impl EventLoop {
                         SwarmEvent::Behaviour(OutEvent::CommunicationError(err)) => {
                             bail!(err.context("Communication error"))
                         }
-                        _ => {}
-                    }
-                },
-                option = self.dial_alice.recv().fuse() => {
-                    if option.is_some() {
-                           let peer_id = self.alice_peer_id;
-                        if self.swarm.pt.is_connected(&peer_id) {
-                            trace!("Already connected to Alice at {}", peer_id);
-                            let _ = self.conn_established.send(peer_id).await;
-                        } else {
-                            debug!("Dialing alice at {}", peer_id);
-                            libp2p::Swarm::dial(&mut self.swarm, &peer_id).context("Failed to dial alice")?;
+                        SwarmEvent::ConnectionEstablished { peer_id, endpoint, .. } if peer_id == self.alice_peer_id => {
+                            tracing::debug!("Connected to Alice at {}", endpoint.get_remote_address());
                         }
+                        SwarmEvent::Dialing(peer_id) if peer_id == self.alice_peer_id => {
+                            tracing::debug!("Dialling Alice at {}", peer_id);
+                        }
+                        SwarmEvent::ConnectionClosed { peer_id, endpoint, num_established, .. } if peer_id == self.alice_peer_id && num_established == 0 => {
+                            bail!("Lost connection to Alice at {}", endpoint.get_remote_address())
```

Comment:
"In case we ever loose the connection to Alice, we simply stop the
event loop to abort the swap execution." Will the stop the event loop and abort swap execution? Won't this prevent bob from recovering if he cannot connect to Alice.